node/actors/bft/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,599 @@
+    assert_ne!(
+        util.view_leader(leader_prepare.view),
+        util.key_at(0).public()
+    );
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+
+    assert_eq!(
+        res,
+        Err(LeaderPrepareError::InvalidLeader {
+            correct_leader: util.key_at(1).public(),
+            received_leader: util.key_at(0).public(),
+        })
+    );
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_sig() {
+    let mut util = Util::make().await;
+
+    let mut leader_prepare = util.make_leader_prepare(None::<fn(&mut LeaderPrepare)>);
+    leader_prepare.sig = util.rng().gen();
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+
+    assert_eq!(
+        res,
+        Err(LeaderPrepareError::InvalidSignature(
+            SignatureVerificationFailure
+        ))
+    );
+}
+
+#[tokio::test]
+async fn leader_prepare_invalid_prepare_qc_different_views() {
+    let mut util = Util::make().await;
+
+    let replica_prepare = util.make_replica_prepare(None::<fn(&mut ReplicaPrepare)>);
+    let res = util.dispatch_replica_prepare(replica_prepare.clone());
+    assert_eq!(res, Ok(()));
+
+    let mut leader_prepare = util.recv_leader_prepare().await.unwrap();
+    leader_prepare.view = leader_prepare.view.next();
+    let leader_prepare = util
+        .own_key()
+        .sign_msg(ConsensusMsg::LeaderPrepare(leader_prepare));
+    let res = util.dispatch_leader_prepare(leader_prepare).await;
+
+    assert_eq!(
+        res,
+        Err(LeaderPrepareError::InvalidPrepareQC(anyhow!(
+            "PrepareQC contains messages for different views!"
+        )))
+    );
+}
+
+// #[tokio::test]
+// async fn replica_commit_sanity() {
+//     let mut util = Util::make().await;
+//
+//     let replica_prepare = util::make_replica_prepare(&consensus, None::<fn(&mut ReplicaPrepare)>);
+//     let res = util::dispatch_replica_prepare(&ctx, &mut consensus, replica_prepare);
+//     assert_eq!(res, Ok(()));
+//     let leader_prepare = util::make_leader_prepare_from_replica_prepare(&consensus, &mut rng, replica_prepare, Some(|msg: &mut LeaderPrepare| {
+//         msg.view = ViewNumber(2);
+//     }));
+//     let replica_commit = util::make_replica_commit(&consensus, leader_prepare.msg., None::<fn(&mut ReplicaCommit)>);
+//
+//     let res = scope::run!(&ctx, |ctx, s| {
+//         s.spawn_blocking(|| {
+//             let res = consensus
+//             .leader
+//             .process_replica_commit(ctx, &consensus.inner, leader_prepare.cast().unwrap());
+//             Ok(res)
+//         })
+//         .join(ctx)
+//     })
+//         .await
+//         .unwrap();
+//
+//     assert_eq!(
+//         res,
+//         Err(LeaderPrepareError::InvalidPrepareQC(anyhow!("PrepareQC contains messages for different views!")))
+//     );
+// }
+
+mod util {
```

Comment:
Inline modules are not very readable; in [our style guide](https://github.com/matter-labs/era-consensus/blob/main/docs/style.md#codebase-layout), we advise always placing modules in separate files.

Reply:
Following https://github.com/matter-labs/era-consensus/pull/39#pullrequestreview-1739967766 I ended up moving it to `testonly`, although it's not being used outside of the module. 
Another option would be to simply remove the inline module, making the struct part of the enclosing unit tests module.

Change here: https://github.com/matter-labs/era-consensus/pull/39/commits/b489f0f90a45719bb0179e09842fe60c8e602112